modules/js/rackandpinion.js

```diff
@@ -1,680 +1,913 @@
 // module indicator and base variables
 rackPinionModule = true;
-var c = 369
-var rectBase = 600
-var originalWidth1
-var originalWidth2
-var newWidth1
-var newWidth2
-module.verticalSpace = 0
+let c = 369; //connector length
+const rectBase = 600;
+let originalWidth1;
+let originalWidth2;
+var newWidth1;
+var newWidth2;
+module.verticalSpace = 0;
 // generate small gear
 ///////////////////////////GEAR SIZE///////////////////////////////////
-function smallGear(){
-  angleFactor = 0.21
-  deleteConstraint(compositeArray[1].bodies[0], compositeArray[0].bodies[0])
-  Body.setPosition(compositeArray[0].bodies[0], {x:compositeArray[0].constraints[0].pointA.x, y:compositeArray[0].constraints[0].pointA.y+130} )
-  Body.setAngle(compositeArray[1].bodies[0], 0)
-  radius = 48;
-  compositeArray[1].radius = radius
-  steps = (0.25 * radius)*2;
+function gear(rad) {
```

Comment:
Could use a better function name: makeGear, constructGear, renderGear or whatever it does :)